abnf/odata-abnf-construction-rules.txt

```diff
@@ -549,7 +549,7 @@ collectionNavigationExpr = collectionNavNoCastExpr
 
 collectionNavNoCastExpr = collectionPathExpr
                         / keyPredicate [ singleNavigationExpr ] 
-                        / filterExpr [ collectionNavigationExpr ]                     
+                      ;  / filterExpr [ collectionNavigationExpr ]                     

```

Comment:
Removing the rule removes the option `collectionNavNoCastExpr = filterExpr keyPredicate`, therefore it is not redundant.

Reply:
A key predicate after a filter does not make sense: just use the key predicate directly, so I would few this as an added benefit 😄

Reply:
Technically, it is an incompatible (if sensible) change to the ABNF that deserves discussion through an OASIS issue.

Reply:
See [ODATA-1507](https://issues.oasis-open.org/browse/ODATA-1507).

Reply:
Thanks, added to the TC agenda

Reply:
TC 2022-03-10 agreed to keep the option `collectionNavNoCastExpr = filterExpr keyPredicate`. I have added a test case, this proves that the rule is not redundant.